src/main/scala/beam/agentsim/agents/modalbehaviors/DrivesVehicle.scala

```diff
@@ -360,27 +364,12 @@ trait DrivesVehicle[T <: DrivingData] extends BeamAgent[T] with Stash {
             if (currentBeamVehicle.isBEV | currentBeamVehicle.isPHEV) {
               stall.chargingPointType match {
                 case Some(_) =>
-                  handleStartCharging(tick, currentBeamVehicle)
-                  if (ChargingPointType.getChargingPointInstalledPowerInKw(
-                        currentBeamVehicle.stall.get.chargingPointType.get
-                      ) > 20.0) {
-                    val (sessionDuration, energyDelivered) =
-                      currentBeamVehicle.refuelingSessionDurationAndEnergyInJoules()
-                    log.debug(
-                      "scheduling EndRefuelSessionTrigger at {} with {} J to be delivered, triggerId: {}",
-                      tick + sessionDuration.toInt,
-                      energyDelivered,
-                      triggerId
-                    )
-                    scheduler ! ScheduleTrigger(
-                      EndRefuelSessionTrigger(
-                        tick + sessionDuration.toInt,
-                        tick,
-                        energyDelivered,
-                        currentBeamVehicle
-                      ),
-                      self
-                    )
+                  if (ChargingPointType
+                        .getChargingPointInstalledPowerInKw(currentBeamVehicle.stall.get.chargingPointType.get) > 20.0) {
+
+                    log.debug("Sending ChargingPlugRequest to chargingNetworkManager at {}", tick)
+                    chargingNetworkManager ! ChargingPlugRequest(currentBeamVehicle, self)
+                    handleStartCharging(tick, currentBeamVehicle)
```

Comment:
Why was this moved inside the if - was it a bug to begin with?

Reply:
It wasn't moved, the if was there before:

![image](https://user-images.githubusercontent.com/427642/94290682-d3933180-ff5a-11ea-90d3-f9114c35de63.png)

Reply:
Not the `if`, the `handleStartCharging` invocation. It was before the `if` and now it is inside the `if`

Reply:
Now I see. And yes, it was moved for a reason the charging can start only when the condition is `true`.
And if to leave the method outside of `if` then when we start handling charging - we emit the matsim event `ChargingPlugInEvent` but no any message will be sent to ChargingNetworkManager, that means charging actually not happens